osquery/tables/system/posix/sudoers.cpp

```diff
@@ -49,50 +51,58 @@ void genSudoersFile(const std::string& filename,
 
   auto lines = split(contents, "\n");
   for (auto& line : lines) {
-    Row r;
-    boost::trim(line);
+    // sudoers uses EBNF for grammer. But for our purposes, we don't need a full
+    // parsing. We're just conveying simplified information. We can just split
+    // it into the leading token and the trailing token.
 
-    // Only add lines that are not comments or blank.
-    if (line.size() > 0 && line.at(0) != '#') {
-      r["source"] = filename;
+    boost::trim_if(line, boost::is_any_of(kSudoWhitespaceChars));
 
-      auto header_pos = line.find_first_of("\t\v ");
-      r["header"] = line.substr(0, header_pos);
+    if (line.size() == 0) {
+      continue;
+    }
 
-      if (header_pos == std::string::npos) {
-        header_pos = line.size() - 1;
-      }
+    // Find the rule header
+    auto header_len = line.find_first_of(kSudoWhitespaceChars);
+    auto header = line.substr(0, header_len);
+    boost::trim_if(header, boost::is_any_of(kSudoWhitespaceChars));
 
-      r["rule_details"] = line.substr(header_pos + 1);
+    // skip comments. We need to manually exlude the includes here. This
+    // duplication allows an early return
+    if (line.at(0) == '#' && header != "#include" && header != "#includedir") {
+      continue;
+    }
 
-      results.push_back(std::move(r));
-    } else if (line.find("#includedir") == 0) {
-      auto space = line.find_first_of(' ');
+    // Find the next field. Instead of skipping the whitespace, we
+    // include it, and then trim it.
+    auto rule_details =
+        (header_len < line.size()) ? line.substr(header_len) : "";
+    boost::trim_if(rule_details, boost::is_any_of(kSudoWhitespaceChars));
+
+    // If an include is _missing_ the target to include, treat it like a
+    // comment.
+    if (rule_details == "" &&
+        (header == "#include" || header == "@include" ||
```

Comment:
Nitpick but if you are checking `header` here and then below, you could optimize

```
auto is_include = (header == "#include" || header == "@include");
auto is_includedir = (header == "#includedir" || header == "@includedir");
```

Then this becomes:

```
if (rule_details.empty() && (is_include || is_include_dir)) {
```

And below is either `if (is_include)` or `if (is_includedir)`.